### ISSUE TITLE
Overlay Popup

### ISSUE THREAD

**Issue by kspearrin** [bitwarden/browser#3]
Implement an overlay popup content script that will assist with autofilling in-line within the website. The popup will overlay the website using shadow-DOM techniques. The overlay will be opened by clicking an icon that is presented with login form fields on the page.

**Comment by joshsleeper** [comment]
Came here just to say I wanted this! Glad to see you're already on top of it. 😁

**Comment by pannal** [comment]
Yes, please!

**Comment by dev-manager-uk** [comment]
Is there an approx ETA for this?  

Lastpass has a little logo in each form input box that you can click and it gives you a list of matching sites to autofill.  Works pretty well.

**Comment by mhammond** [comment]
FTR, I moved to bitwarden from lastpass precisely because it *does not* have this feature - too many sites struck edge-cases which caused lastpass to slow the browser down and it wasn't uncommon for internal mozilla conversations to go something like:

a) oh man, why does [irccloud/slack/etc] slow Firefox down so horribly?
b) I bet you have lastpass installed - try disabling it?
a) YES! Thank you!

I'm perfectly happy clicking the button before bitwarden tries to be too smart for its own good.

**Comment by pannal** [comment]
I second that, and wasn't the more DOM-intrusive approach of LastPass one of the reasons for the recent attacks/security flaws?

**Comment by dev-manager-uk** [comment]
Speed and Security.  Two excellent reasons not to add that feature.  I bow to you both.

A menu right next to forms could be a good solution.  But I guess we're only talking about a bit of mouse travel time, so maybe not a priority...

**Comment by Mentis** [comment]
Can we make this optional then? It is really helpfull to have this feature in mobile browsers like Firefox for Android which does not support Right-Click menu.

**Comment by poopledoople** [comment]
This feature is what I'm sorely missing from Lastpass. It makes logging in so much easier since your cursor doesn't have to travel such a far distance. 

Keyboard shortcuts aren't for everybody.

**Comment by echosa** [comment]
> Keyboard shortcuts aren't for everybody.

I agree. While I don't mind using keyboard shortcuts, there are several friends and family members I wouldn't mind moving away from LastPass. However, without these convenience buttons on the forms themselves, that will be a huge headache. They're used to clicking on that icon for LastPass, and habits are hard to break. 

I'd definitely love to see this feature implemented. It might seem minor, but trust it, this would be a *huge* boon for Bitwarden.

**Comment by arnabuchiha** [comment]
Will this feature be implemented ever? People who don't want this feature should have the option to disable.
